### ISSUE TITLE
code-block width changes with :linenos:

### ISSUE THREAD

**Issue by andreacassioli** [sphinx-doc/sphinx#2427]
I have notice that `code-block` layout changes depending on whether line numbers are included. Indeed the text background width is shorter with numbers.

Try for instance

```
.. code-block:: python
   :caption: test
   :lineos:

   import sys

```

The result is 
![screenshot from 2016-04-06 10 39 17](https://cloud.githubusercontent.com/assets/6704112/14310745/e9236094-fbe3-11e5-8cbd-c7aa08f0f0d3.png)

while with

```
.. code-block:: python
   :caption: test

   import sys

```

is 

![screenshot from 2016-04-06 10 41 02](https://cloud.githubusercontent.com/assets/6704112/14310780/12cd2416-fbe4-11e5-8a63-3d866d36fb9f.png)

 That's quite annoying because I may want to have some code blocks with line numbers, but other without: the overall page layout is then a mess.

**Comment by andreacassioli** [comment]
BTW: it is not just the width but the layout as well that changes.

**Comment by andreacassioli** [comment]
Any clue? Is it a docutils issue?

**Comment by yasoob** [comment]
This is also a big issue when generating a PDF. Any update on this?

**Comment by mgeier** [comment]
I think the original CSS issue has been fixed in #7482.

@yasoob What exactly is the issue with PDF? Probably that's a separate problem? Maybe you should create a separate issue for that?

**Comment by yasoob** [comment]
@mgeier you are absolutely correct. My issue is separate from this. I realized that later on. If you think this one has been solved in #7482 then then is good to close :)
